Add RegularPolygonPixelRegion class

larrybradley opened the pull request:
This PR adds a `RegularPolygonPixelRegion` class defined by a center, number of vertices (sides), radius, and optional angle.

Examples:

![regions-RegularPolygonPixelRegion-1](https://user-images.githubusercontent.com/4992897/128442936-cb1e9864-a49a-4d65-93b3-cb6b6e866573.png)

keflavich commented:
Does this region type have any round-trippable serializations?  I don't think ds9 or CRTF support it, but perhaps the VO serialization format does?  If not, do we have a list of regions that can('t) be round-trip-serialized?  We should.  This is probably something to raise as a separate issue.

larrybradley commented:
Serialization for this object doesn't work yet, but `RegularPolygonPixelRegion` is subclassed from `PolygonPixelRegion` so I can serialize/write it to ds9 using the polygon shape.  Reading/writing it from ds9 would then return a `PolygonPixelRegion` instead of a `RegularPolygonPixelRegion` -- so not exactly round-trippable to give the same *object*, but the defined region would be the same.

keflavich commented:
Yes, that's exactly what I meant - a user can serialize it just fine, but they will not get back a similarly-manipulatable object if they read it from ds9/carta.

larrybradley commented:
We could always define our own serialization format (perhaps using ASDF?), but that's not a priority.

keflavich commented:
I'm not suggesting [that](https://xkcd.com/927/)! 😱   I'm just saying we should note it!

larrybradley commented:
We *could* also add a method converts PolygonPixelRegion to RegularPolygonPixelRegion (if it's a regular polygon).

larrybradley commented:
I've fixed this serialization and added a note in the docstring about the round-trip serialization:

<img width="629" alt="Screen Shot 2021-08-05 at 10 59 42 PM" src="https://user-images.githubusercontent.com/4992897/128449578-96f3405e-2549-495a-b27e-025e8422b84a.png">

keflavich commented:
For the regular polygon 'recognizer', we'd be subject to precision limitations in the serialization, which would mean probably introducing some threshold for 'regular'.  That's fine, just awkward.
